backend/rates/coingecko.go

```diff
@@ -0,0 +1,153 @@
+package rates
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"sort"
+	"strconv"
+	"time"
+
+	"github.com/digitalbitbox/bitbox-wallet-app/util/observable"
+	"github.com/digitalbitbox/bitbox-wallet-app/util/observable/action"
+)
+
+const coingeckoAPIV3 = "https://api.coingecko.com/api/v3"
+
+// forwardHistoryLoop periodically updates historical market exchange rates
+// forward in time starting with the last fetched timestamp.
+func (updater *RateUpdater) forwardHistoryLoop() {
+	for {
+		// When to update next, after this loop iteration is done.
+		untilNext := 10 * time.Minute // TODO: add jitter
```

Comment:
jitter: why?

Reply:
because we'll be fetching historical data for many coins, right: BTC, LTC, ETH and so on.
so, we don't want to hammer the API with 5-10 QPS every 10min exactly. the idea is to be more gentle and spread the load on the API provider over a couple minutes.

Reply:
and it'll work in our favour regarding the API rate limits